### PR TITLE
feat: comprehensive fuzzing for BTreeMap

### PR DESCRIPTION
**Pull request by ielashi**
Adds a fuzz test that can run indefinitely comparing the results of a stable BTreeMap to a standard BTreeMap.

I ended up using `proptest` rather than the conventional `cargo fuzz` for two reasons:

1. `cargo fuzz` requires nightly rust.
2. With proptest, we can have a short version of the fuzz test that runs as part of CI (which is included in this commit).

### CODE REVIEW

File: src/btreemap/proptests.rs [hunk 1]
```diff
@@ -1,17 +1,89 @@
...
+#[proptest(cases = 10)]
+fn comprehensive(#[strategy(pvec(op_strategy(), 100..5_000))] ops: Vec<Operation>) {
+    let mem = make_memory();
+    let mut btree = BTreeMap::new(mem);
+    let mut std_btree = StdBTreeMap::new();
+
+    // Execute all the operations, validating that the stable btreemap behaves similarly to a std
+    // btreemap.
+    for op in ops.into_iter() {
+        execute_operation(&mut std_btree, &mut btree, op);
```
Comment (AlexandraZapuc) [inline src/btreemap/proptests.rs:52]: nit: I would rename this function. Maybe `validate_trees_after_operation`? Or something else that makes it obvious the purpose of the function is to verify the trees.

File: src/btreemap/proptests.rs [hunk 2]
```diff
@@ -1,17 +1,89 @@
...
+    Insert { key: Vec<u8>, value: Vec<u8> },
+    Iter { from: usize, len: usize },
+    Get(usize),
+    Remove(usize),
+}
+
+// A custom strategy that gives unequal weights to the different operations.
+// Note that `Insert` has a higher weight than `Remove` so that, on average, BTreeMaps
+// are growing in size the more operations are executed.
+fn op_strategy() -> impl Strategy<Value = Operation> {
```
Comment (AlexandraZapuc) [inline src/btreemap/proptests.rs:26]: nit: rename `operation_strategy`

File: src/btreemap/proptests.rs [hunk 3]
```diff
@@ -1,17 +1,89 @@
...
+    let mut runner = TestRunner::default();
+
+    let mem = make_memory();
+    let mut btree = BTreeMap::new(mem);
+    let mut std_btree = StdBTreeMap::new();
+
+    let mut i = 0;
+
+    loop {
+        let op = op_strategy().new_tree(&mut runner).unwrap().current();
```
Comment (AlexandraZapuc) [inline src/btreemap/proptests.rs:77]: nit: Rename `op` to `operation`.

**Review by AlexandraZapuc — APPROVED** [review]

File: src/btreemap/proptests.rs (hunk 1 above)
Comment (ielashi) [reply src/btreemap/proptests.rs:52]: Actually, the purpose of this method _is_ to execute an operation, so the current naming is accurate.

**Pull request merged**